File: Desarrollo Software-DS/Practica 1/Sesion3/Opcional/dart/lib/parteTecnico.dart
import 'dart:io';
import 'dart:math';
import 'dart:async';

class parteTecnico {
  int notaTrabajo = 0;
  int actitud = 0;
  bool recomendacion = false;

  parteTecnico(int nt, int act, bool rc) {
    this.notaTrabajo = nt;
    this.actitud = act;
    this.recomendacion = rc;
  }

  int getnotaTrabajo() {
    return notaTrabajo;
  }

  int getActitud() {
    return actitud;
  }

  bool getRecomendacion() {
    return recomendacion;
  }

  void rellenarParteTecnico() {
    print("\nEl cliente ha rellenado el parte con los siguientes datos\n");

    Random random = new Random();

    this.notaTrabajo = random.nextInt(11);
    print("Ha valorado la nota del trabajo con un " + this.notaTrabajo.toString() + "\n");

    this.actitud = random.nextInt(11);
    print("Ha valorado la actitud con un " + this.actitud.toString() + "\n");

    int resp = random.nextInt(2);

    if (resp == 0) {
      this.recomendacion = false;
      print("No ha recomendado al técnico\n");
    } else {
      this.recomendacion = true;
      print("Ha recomendado al técnico");
    }
  }
}

File: Tercero/Desarrollo Software-DS/Practica 1/Sesion3/Obligatoria/S3/parteTecnico.cpp
#include"parteTecnico.h"


parteTecnico::parteTecnico(int nt,int act,bool rc)
{
    this->notaTrabajo=nt;
    this->actitud=act;
    this->recomendacion=rc;
}


void parteTecnico::rellenarParteTecnico()
{   
    cout<<"\nVamos a rellenar el Parte del Tecnico"<<endl<<endl;
    int nT;
    do{
    cout<<"Introduzca una nota del trabajo efectuado por el tecnico(0-10):"<<endl;
    cin>>nT;
    }while(nT<0 || nT>10);
    
    int act;
    do{
    cout<<"Introduzca la actitud que ha tenido el tecnico(0-10):"<<endl;
    cin>>act;
    }while(act<0 || act>10); 
    
    string resp;
    string si="si";
    string Si="Si";
    string no="no";
    string No="No";
    bool rec;
    do{
    cout<<"¿Recomiendas al técnico?(Si o No):"<<endl;
    cin>>resp;
    }while(resp.compare(si) && resp.compare(Si) && resp.compare(no) && resp.compare(No));

    
    if(resp=="si" || resp=="Si")
    {
        
        rec=true;
    }
    else
    {
        rec=false;
    }

    
    this->notaTrabajo=nT;
    this->actitud=act;
    this->recomendacion=rec;
    
}

